crates/polars-core/src/chunked_array/ops/rolling_window.rs

```diff
@@ -111,6 +111,12 @@ mod inner_mod {
                         // we are in bounds
                         let arr_window = unsafe { arr.slice_typed_unchecked(start, size) };
 
+                        // ensure we still meet window size criteria after removing null values
+                        if size - arr_window.null_count() < options.min_periods {
```

Comment:
I realize that the null count can probably more efficiently be tracked by only tracking the head and tail of each new window, but because this is the slow `rolling_map` this is almost guaranteed not to be a bottleneck.

Reply:
Nah.. Indeed. This might be the worst code in the repo. :(